search-app/src/cmr/search/services/parameters/parameter_validation.clj

```diff
@@ -789,3 +798,62 @@
                         deleted-colls-revision-date-validation])]
     (when (seq errors)
       (errors/throw-service-errors :bad-request errors))))
+
+(defn- deleted-grans-result-format-validation
+  "Validates that the only result format support by deleted granules search is :xml"
+  [params]
+  (when-not (= :json (:result-format params))
+    [(format (str "Result format [%s] is not supported by deleted granules search. "
+                  "The only format that is supported is json")
+             (name (:result-format params)))]))
+
+(defn- unrecognized-deleted-grans-params-validation
+  "Validates that no invalid parameters were supplied to deleted granules search"
+  [params]
+  (map #(format "Parameter [%s] was not recognized." (csk/->snake_case_string %))
+       (set/difference (set (keys params)) valid-deleted-granules-search-params)))
+
+(defn- validate-deleted-grans-revision-date-str
+  [revision-date]
+  (when (.contains revision-date ",")
+    [(format (str "Invalid format for revision date, only a starting date is allowed "
+                  "for deleted granules search, but was [%s]") revision-date)]))
+
+(defn- deleted-grans-revision-date-validation
+  "Validates that deleted time can only have a start date for deleted collections search"
+  [params]
+  (when-let [revision-date (or (:revision-date params)
+                               (:revision_date params))]
+    (if (sequential? revision-date)
+      (if (> (count revision-date) 1)
+        [(format "Only one deleted time is allowed, but %s were provided." (count revision-date))]
+        (validate-deleted-grans-revision-date-str (first revision-date)))
+      (validate-deleted-grans-revision-date-str revision-date))))
+
+(defn- deleted-grans-revision-date-range-validation
+  "Validates that deleted time can only have a start date for deleted collections search"
+  [params]
+  (when-let [revision-date (or (:revision-date params)
+                               (:revision_date params))]
+    (when (t/before? (dt-parser/parse-datetime revision-date) (t/minus- (t/now) (t/days 365)))
+      [(format "Revision date must be withing one year of today.")])))
+
+(defn- deleted-grans-revision-required-validation
+  "Validates that deleted time can only have a start date for deleted collections search"
+  [params]
+  (let [revision-date (or (:revision-date params)
+                          (:revision_date params))]
+    (when (not (seq revision-date))
+      [(format "One revision date is required for deleted granules search.")])))
+
+(defn validate-deleted-granules-params
+  "Validates the query parameters passed in with deleted collections search.
+   Throws exceptions to send to the user if a validation fails."
+  [params]
+  (let [errors (mapcat #(% params)
+                       [unrecognized-deleted-grans-params-validation
```

Comment:
Were there any validations from deleted-collections that could be reused? I was just thinking any of the date parsing, single value validations should have been identical to the collection ones.